solr/core/src/java/org/apache/solr/handler/component/ShardDoc.java

```diff
@@ -56,6 +63,25 @@ public ShardDoc() {
     super(-1, Float.NaN);
   }
 
+  public void setState(Map<String, Object> state) {
```

Comment:
Is this basically just a way to encode and decode a `ShardDoc` to something that is trivially serializable using JavaBin? If so... should this instead be a direct encode/decode here instead?

The client can then base64 encode as needed.

Not sure we need this interstitial map representation anywhere.